### PR TITLE
Feat report googlechat (#1257)

### LINKED ISSUE #1257: reporting to Google Chat
I would like to see vuls reporting on Google Chat ( [chat.google.com](url) ).
NOTE: Google Hangout is an ancient product.

### PR DESCRIPTION
**Pull request by shigechika**
If this Pull Request is work in progress, Add a prefix of “[WIP]” in the title.

# What did you implement:

Added new option "-to-googlechat" in vuls report.
"vuls report -to-googlechat" sent report Google Chat

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. 

Fixes #1257

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update → https://github.com/vulsdoc/vuls/pull/155

# How Has This Been Tested?

vuls-report-googlechat
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

# Checklist:
You don't have to satisfy all of the following.

- [x] Write tests
- [x] Write documentation
- [x] Check that there aren't other open pull requests for the same issue/feature
- [x] Format your source code by `make fmt`
- [x] Pass the test by `make test`
- [x] Provide verification config / commands
- [x] Enable "Allow edits from maintainers" for this PR
- [x] Update the messages below

***Is this ready for review?:*** NO  

# Reference

* https://blog.github.com/2015-01-21-how-to-write-the-perfect-pull-request/

### CODE REVIEW

**Comment by shigechika** [comment]
output format updated. append diff, attack vector and exploits info.
スクリーンショット 2021-06-24 12 14 18

File: config/googlechatconf.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+// Validate validates configuration
+func (c *GoogleChatConf) Validate() (errs []error) {
+	if !c.Enabled {
+		return
+	}
+	if len(c.WebHookURL) == 0 {
+		errs = append(errs, xerrors.New("googleChatConf.webHookURL must not be empty"))
+	}
+	_, err := govalidator.ValidateStruct(c)
```
Comment (kotakanbe) [inline config/googlechatconf.go:24]: Stop calling ValidateStruct because the struct tag does not have a validation rule .

**Review by kotakanbe — CHANGES_REQUESTED** [review]

File: subcmds/discover.go [hunk 2]
```diff
@@ -157,6 +157,12 @@ func printConfigToml(ips []string) (err error) {
 #room     = "xxxxxxxxxxx"
 #apiToken = "xxxxxxxxxxxxxxxxxx"
 
+# https://vuls.io/docs/en/config.toml.html#googlechat-section
+#[googlechat]
+#webHookURL = "https://chat.googleapis.com/v1/spaces/xxxxxxxxxx/messages?key=yyyyyyyyyy&token=zzzzzzzzzz%3D"
+#skipHealthy = false
+#filter = "(^(\\[Reboot Required\\] )?(?:(spam|ham)).*|.*(?:egg)$)" # exclude spamonigiri, hamburger, boiledegg
```
[the remark this answers is not in the document]
Comment (shigechika) [reply subcmds/discover.go:164]: This regex can include/exclude by ServerName.
I think `Ignore` has excluding image. so `IgnoreServerNameRegexp` seems best naming.
How about `ServerNameRegexp` ?

File: reporter/googlechat.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+			r.ServerInfo(),
+			r.ScannedCves.FormatCveSummary(),
+			r.ScannedCves.FormatFixedStatus(r.Packages),
+			r.FormatUpdatablePkgsSummary())}
+		for _, vinfo := range r.ScannedCves.ToSortedSlice() {
+			max := vinfo.MaxCvssScore().Value.Score
+
+			exploits := ""
+			if 0 < len(vinfo.Exploits) || 0 < len(vinfo.Metasploits) {
+				exploits = "*POC*"
```
[the remark this answers is not in the document]
Comment (shigechika) [reply reporter/googlechat.go:41]: Okey, I'll rename.

File: subcmds/discover.go [hunk 4]
```diff
@@ -157,6 +157,12 @@ func printConfigToml(ips []string) (err error) {
 #room     = "xxxxxxxxxxx"
 #apiToken = "xxxxxxxxxxxxxxxxxx"
 
+# https://vuls.io/docs/en/config.toml.html#googlechat-section
+#[googlechat]
+#webHookURL = "https://chat.googleapis.com/v1/spaces/xxxxxxxxxx/messages?key=yyyyyyyyyy&token=zzzzzzzzzz%3D"
+#skipHealthy = false
```
[the remark this answers is not in the document]
Comment (shigechika) [reply subcmds/discover.go:163]: OK, I'll rename.

File: config/googlechatconf.go (hunk 1 above)
Comment (shigechika) [reply config/googlechatconf.go:24]: I understood about govalidator. I rewrite.

**Review by kotakanbe — APPROVED** [review]

**Comment by kotakanbe** [comment]
Many thanks :)

**Pull request merged**